agent/api/ecsclient/client.go

```diff
@@ -123,21 +123,31 @@ func (client *APIECSClient) RegisterContainerInstance(containerInstanceArn strin
 
 func (client *APIECSClient) registerContainerInstance(clusterRef string, containerInstanceArn string, attributes []string) (string, error) {
 	registerRequest := ecs.RegisterContainerInstanceInput{Cluster: &clusterRef}
+	var registrationAttributes []*ecs.Attribute
 	if containerInstanceArn != "" {
+		// We are re-connecting a previously registered instance, restored from snapshot.
 		registerRequest.ContainerInstanceArn = &containerInstanceArn
+	} else {
+		// This is a new instance, not previously registered.
+		// Custom attribute registration only happens on initial instance registration.
```

Comment:
Why does custom attribute registration only happen on initial registration?  Seems like it would be reasonable for this to happen on every registration, same as the other attributes.